src/main/java/org/zeplinko/commons/lang/ext/util/Retryable.java

```diff
@@ -306,40 +345,74 @@ private MaxRetriesExceededException(String message) {
 
     /**
      * Immutable value object describing the outcome of a {@link Retryable#run()}
-     * call.
+     * call. This class extends {@link AbstractOutcome} to provide common outcome
+     * functionality with success/failure semantics.
      */
     @Getter
-    @RequiredArgsConstructor
-    public static final class Outcome<T> {
+    public static final class Outcome<T> extends AbstractOutcome<T, Exception> {
+
+        /**
+         * Enumeration of possible termination reasons for retry execution.
+         */
         public enum TerminationReason {
+            /**
+             * The task succeeded and result validation passed.
+             */
             SUCCEEDED,
+            /**
+             * Maximum retries were exceeded due to result validation failures.
+             */
             RETRIES_EXHAUSTED_INVALID_RESULT,
+            /**
+             * Maximum retries were exceeded due to retryable exceptions.
+             */
             RETRIES_EXHAUSTED_RETRYABLE_EXCEPTION,
+            /**
+             * Execution was aborted due to a non-retryable exception.
+             */
             ABORTED_NON_RETRYABLE_EXCEPTION,
+            /**
+             * Execution was aborted due to an exception matching the no-retry predicate.
+             */
             ABORTED_SKIP_RETRY_EXCEPTION,
+            /**
+             * Execution was interrupted while waiting between attempts.
+             */
             INTERRUPTED
         }
 
-        private final boolean success;
-
         private final int attempts;
 
-        private final T data;
-
-        private final Exception error;
-
         private final TerminationReason reason;
 
+        Outcome(int attempts, T data, Exception error, TerminationReason reason) {
+            super(data, error);
+            this.attempts = attempts;
+            this.reason = reason;
+        }
+
         static <T> Outcome<T> success(T data, int attempts) {
-            return new Outcome<>(true, attempts, data, null, TerminationReason.SUCCEEDED);
+            return new Outcome<>(attempts, data, null, TerminationReason.SUCCEEDED);
         }
 
         static <T> Outcome<T> failure(Exception error, int attempts, TerminationReason reason) {
-            return new Outcome<>(false, attempts, null, error, reason);
+            return new Outcome<>(attempts, null, error, reason);
         }
 
+        /**
+         * Converts this outcome to a {@link Result}.
+         *
+         * @return a {@link Result} representing the success or failure of the retry
+         *         operation
+         */
         public Result<T, Exception> toResult() {
-            return success ? Result.success(data) : Result.failure(error);
+            return Objects.nonNull(getData()) ? Result.success(getData()) : Result.failure(getError());
```

Comment:
Use the isFailure() method here to decide whether or not to return success or failure instance of Result

Reply:
done